op-supervisor/supervisor/backend/db/logs/db.go

```diff
@@ -307,8 +307,8 @@ func (db *DB) Contains(blockNum uint64, logIdx uint32, logHash common.Hash) (typ
 		panic("expected iterator to stop with error")
 	}
 	if errors.Is(err, types.ErrFuture) {
-		// Log is known, but as part of an unsealed block.
-		return types.BlockSeal{}, nil
+		// Log is known, but as part of an unsealed block (we did not reach the block number yet).
```

Comment:
We can predict this fact by checking `db.lastEntryContext.blockNum` against the requested `blockNum`, I think. That should avoid traversal work when the block isn't sealed in the DB yet, which I imagine could be a pretty hot path, as the most interesting requests will be at the tip of the db.

Reply:
Good idea